scripts/pre_commit_linter.py

```diff
@@ -463,8 +470,237 @@ def _redirect_stdout(new_target):
         sys.stdout = old_target
 
 
+def _get_all_files():
```

Comment:
Please add proper comments to this function to explain its working.

Reply:
Done